plugins/Web/WebBackend.vala

```diff
@@ -0,0 +1,176 @@
+    b.ref ();
+    return b;
+}
+
+public static Maya.Backend backend;
+
+public class Maya.WebBackend : GLib.Object, Maya.Backend {
+
+    public WebBackend () {
+        backend = this;
+    }
+
+    public string get_name () {
+        return _("On the web");
+    }
+
+    public string get_uid () {
+        return "webcal-stub";
+    }
+
+    public Gee.Collection<PlacementWidget> get_new_calendar_widget (E.Source? to_edit = null) {
+        var collection = new Gee.LinkedList<PlacementWidget> ();
+
+        bool keep_copy = false;
+        if (to_edit != null) {
+            E.SourceOffline source_offline = (E.SourceOffline)to_edit.get_extension (E.SOURCE_EXTENSION_OFFLINE);
+            keep_copy = source_offline.stay_synchronized;
+        }
+
+        collection.add (Maya.DefaultPlacementWidgets.get_keep_copy (0, keep_copy));
+
+        var url_label = new PlacementWidget ();
+        url_label.widget = new Gtk.Label (_("URL:"));
+        ((Gtk.Label) url_label.widget).expand = true;
+        ((Gtk.Misc) url_label.widget).xalign = 1.0f;
+        url_label.row = 1;
+        url_label.column = 0;
+        url_label.ref_name = "url_label";
+        collection.add (url_label);
+
+        var url_entry = new PlacementWidget ();
+        url_entry.widget = new Gtk.Entry ();
+        ((Gtk.Entry)url_entry.widget).text = "http://";
+        url_entry.row = 1;
+        url_entry.column = 1;
+        url_entry.ref_name = "url_entry";
+        url_entry.needed = true;
+        collection.add (url_entry);
+        if (to_edit != null) {
+            E.SourceWebdav webdav = (E.SourceWebdav)to_edit.get_extension (E.SOURCE_EXTENSION_WEBDAV_BACKEND);
+#if HAS_EDS_3_46
+            var uri = webdav.dup_uri ();
+#else
+            var uri = webdav.dup_soup_uri ();
+#endif
+            if (uri.get_port () != 80) {
+                ((Gtk.Entry)url_entry.widget).text = "%s://%s:%u%s".printf (uri.get_scheme (), uri.get_host (), uri.get_port (), uri.get_path ());
+            } else {
+                ((Gtk.Entry)url_entry.widget).text = "%s://%s%s".printf (uri.get_scheme (), uri.get_host (), uri.get_path ());
+            }
+        }
+
+        var secure_checkbutton = new PlacementWidget ();
+        secure_checkbutton.widget = new Gtk.CheckButton.with_label (_("Use a secure connection"));
+        secure_checkbutton.row = 3;
+        secure_checkbutton.column = 1;
+        secure_checkbutton.ref_name = "secure_checkbutton";
+        collection.add (secure_checkbutton);
+        if (to_edit != null) {
+            E.SourceSecurity security = (E.SourceSecurity)to_edit.get_extension (E.SOURCE_EXTENSION_SECURITY);
+            ((Gtk.CheckButton)secure_checkbutton.widget).active = security.secure;
+        }
+
+        return collection;
+    }
+
+    public void add_new_calendar (string name, string color, bool set_default, Gee.Collection<PlacementWidget> widgets) {
+        try {
+            var new_source = new E.Source (null, null);
```

Comment:
For example here you can also construct with properties (at least the display name, I think parent is not possible)